store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -152,35 +171,62 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
                 }
 
                 // validate url is empty or not
                 if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getUrl())) {
-                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_URL);
+                }
+
+                // validate format of url
+                if (!volInfoRequest.getVolumeExternalOpenIOInfo().getUrl().contains("http:/") && !volInfoRequest.getVolumeExternalOpenIOInfo().getUrl().contains("https:/")) {
+                    throw VolumeServiceException.INVALID_REQUEST("Url format is incorrect", VolumeServiceException.BAD_VOLUME_URL);
                 }
 
                 // validate account is empty or not
                 if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getAccount())) {
-                    throw VolumeServiceException.INVALID_REQUEST("Account can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Account can't be null", VolumeServiceException.BAD_VOLUME_ACCOUNT);
                 }
 
+                // validate the accountname ( allowed alpha numeric & _ & - )
+                checkNames(volInfoRequest.getVolumeExternalOpenIOInfo().getAccount(),"AccountName");
+
                 // validate namespace is empty or not
                 if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getNameSpace())) {
-                    throw VolumeServiceException.INVALID_REQUEST("Name Space can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                    throw VolumeServiceException.INVALID_REQUEST("Name Space can't be null", VolumeServiceException.BAD_VOLUME_NAME_SPACE);
                 }
+
+                // validate the nameSpace ( allowed alpha numeric & _ & - )
+                checkNames(volInfoRequest.getVolumeExternalOpenIOInfo().getNameSpace(),"NameSpace");
+
             } else {
                 throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3 or OpenIO",
                         VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
             }
         }
     }
 
+    /**
+     * Validation for openIO AccountName & NameSpace
+     * @return
+     * @throws ServiceException
+     */
+    private static void checkNames(String name , String type) throws ServiceException {
+        String nameToLower = name.toLowerCase();
+        char charArr[] = nameToLower.toCharArray();
+        for (int i=0;i<charArr.length;i++){
+          if ((charArr[i] < 97 || charArr[i] > 122) && charArr[i] != 45 && charArr[i] != 95 && (charArr[i] < 48 || charArr[i] > 57)){
```

Comment:
can't we use https://docs.oracle.com/javase/7/docs/api/java/lang/Character.html#isLetterOrDigit(char) ?

Reply:
removed as this validation was already there

Reply:
@rupeshtelus you can use similar regex for validating URLs https://github.com/Zimbra/zm-hsm-store/blob/develop/src/java/com/zimbra/cs/admin/util/S3BucketGlobalConfigUtil.java#L47

Reply:
changed & pushed